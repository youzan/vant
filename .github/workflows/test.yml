name: CI

on:
  push:
    branches:
      - '**'

  pull_request:
    branches:
      - main

  workflow_dispatch:

jobs:
  lint:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4

      - name: Install pnpm
        run: |
          npm install -g corepack@latest --force 
          corepack enable

      - uses: actions/setup-node@v4
        with:
          node-version: '22.14.0'
          cache: 'pnpm'

      - name: Install dependencies
        run: pnpm install

      - name: Run linter
        run: npm run lint

  test:
    runs-on: macos-14
    steps:
      - uses: actions/checkout@v4

      - name: Install pnpm
        run: |
          npm install -g corepack@latest --force 
          corepack enable

      - uses: actions/setup-node@v4
        with:
          node-version: '22.14.0'
          cache: 'pnpm'

      - name: Install dependencies
        run: pnpm install

      - name: Run test cases
        run: pnpm run test:coverage

      - name: Upload coverage to Codecov
        uses: codecov/codecov-action@v5
        with:
          token: ${{ secrets.CODECOV_TOKEN }}
          directory: ./packages/vant/test/coverage

  build:
    runs-on: macos-14
    steps:
      - uses: actions/checkout@v4

      - name: Install pnpm
        run: |
          npm install -g corepack@latest --force 
          corepack enable

      - uses: actions/setup-node@v4
        with:
          node-version: '22.14.0'
          cache: 'pnpm'

      - name: Install dependencies
        run: pnpm install

      - name: Build
        run: npm run build
