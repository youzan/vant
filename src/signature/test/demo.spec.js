import Demo from '../demo';
import { snapshotDemo } from '../../../test/demo';

snapshotDemo(Demo);
