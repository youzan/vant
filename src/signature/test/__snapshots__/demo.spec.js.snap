// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders demo correctly 1`] = `
<div>
  <div>
    <div role="switch" aria-checked="true" class="van-switch van-switch--on">
      <div class="van-switch__node"></div>
    </div>
  </div>
  <div>
    <div role="switch" aria-checked="true" class="van-switch van-switch--on van-switch--disabled">
      <div class="van-switch__node"></div>
    </div>
  </div>
  <div>
    <div role="switch" aria-checked="true" class="van-switch van-switch--on van-switch--loading">
      <div class="van-switch__node">
        <div class="van-loading van-loading--circular van-switch__loading"><span class="van-loading__spinner van-loading__spinner--circular"><svg viewBox="25 25 50 50" class="van-loading__circular"><circle cx="50" cy="50" r="20" fill="none"></circle></svg></span></div>
      </div>
    </div>
  </div>
  <div>
    <div role="switch" aria-checked="true" class="van-switch van-switch--on" style="font-size: 24px;">
      <div class="van-switch__node"></div>
    </div>
  </div>
  <div>
    <div role="switch" aria-checked="true" class="van-switch van-switch--on" style="background-color: rgb(238, 10, 36);">
      <div class="van-switch__node"></div>
    </div>
  </div>
  <div>
    <div role="switch" aria-checked="true" class="van-switch van-switch--on">
      <div class="van-switch__node"></div>
    </div>
  </div>
  <div>
    <div class="van-cell van-cell--center">
      <div class="van-cell__title"><span>标题</span></div>
      <div role="switch" aria-checked="true" class="van-switch van-switch--on" style="font-size: 24px;">
        <div class="van-switch__node"></div>
      </div>
    </div>
  </div>
</div>
`;
